taxi/views.py

```diff
@@ -1,47 +1,52 @@
 from django.shortcuts import render
 from django.views import generic
-
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
 from .models import Driver, Car, Manufacturer
 
 
+@login_required
 def index(request):
     """View function for the home page of the site."""
 
     num_drivers = Driver.objects.count()
     num_cars = Car.objects.count()
     num_manufacturers = Manufacturer.objects.count()
+    num_visits = request.session.get("num_visits", 0)
+    request.session["num_visits"] = num_visits + 1
```

Comment:
Ensure that the session middleware is enabled in your Django settings to use session variables like `num_visits`.